src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -279,6 +280,29 @@ export class SavedObjectsRepository {
       }
     }
 
+    let savedObjectWorkspaces = workspaces;
+
+    if (id && overwrite && workspaces) {
+      let currentItem;
+      try {
+        currentItem = await this.get(type, id);
+      } catch (e) {
+        // this.get will throw an error when no items can be found
```

Comment:
Do we want to log the exception?

Reply:
It is better to but for repository there is not log method, maybe we should skip it for now.